Lib/test/test_re.py

```diff
@@ -1784,6 +1784,37 @@ def test_match_repr(self):
         )
         self.assertRegex(repr(second), pattern)
 
+        # Issue 39949: Verify truncation of long match
+        for matched_len in range(30,70):
+            for test, search in ((b"aoeui", rb"\w*"), ("abcde", r"\w*")):
+                string = (test * 20)[:matched_len]
+                match = re.match(search, string)
+
+                displayed = repr(string)
+                if len(displayed) > 50:
+                    # Note: no closing quote on the match.
+                    displayed = displayed[:47] + "..."
+
+                pattern = r"<(%s\.)?%s object; span=\(0, %d\), match=%s>" % (
+                    type(match).__module__, type(match).__qualname__,
+                    matched_len, displayed
+                )
+                self.assertRegex(repr(match), pattern)
+
+        # Issue 39949
+        self.assertEqual(
+            repr(re.match(r'\w*', ('abcde'*10)[:48])),
+            "<re.Match object; span=(0, 48), match='abcdeabcdeabcdeabcdeabcdeabcdeabcdeabcdeabcdeabc'>")
+        self.assertEqual(
+            repr(re.match(r'\w*', ('abcde'*10)[:49])),
+            "<re.Match object; span=(0, 49), match='abcdeabcdeabcdeabcdeabcdeabcdeabcdeabcdeabcdea...>")
+        self.assertEqual(
+            repr(re.match(rb'\w*', (b'aoeui'*10)[:47])),
+            "<re.Match object; span=(0, 47), match=b'aoeuiaoeuiaoeuiaoeuiaoeuiaoeuiaoeuiaoeuiaoeuiao'>")
+        self.assertEqual(
+            repr(re.match(rb'\w*', (b'aoeui'*10)[:48])),
+            "<re.Match object; span=(0, 48), match=b'aoeuiaoeuiaoeuiaoeuiaoeuiaoeuiaoeuiaoeuiaoeui...>")
```

Comment:
Now that we have an explicit test we can see that there is a small issue ;)  the closing `'` gets discarded, it should be `"<re.Match object; span=(0, 48), match=b'aoeuiaoeuiaoeuiaoeuiaoeuiaoeuiaoeuiaoeuiaoeui...'>")`.

I don't think the tests from 1787 to 1802 will be useful since there is those.

Reply:
I have removed other tests.

I'm not sure what to do about `b'abc...'` vs `b'abc...`
`b'abc...` is what the existing code does
but 
`b'abc'...` maybe makes it more clear it's been truncated

Reply:
Like I said in https://github.com/python/cpython/pull/20922#discussion_r441158659 if you use `reprlib` it will take care of `...` for all the various cases.